pkg/rule/rule.go

```diff
@@ -113,7 +113,10 @@ func (r RuleGroup) MarshalYAML() (interface{}, error) {
 func (m *Managers) Update(dataDir string, evalInterval time.Duration, files []string) error {
 	var (
 		errs     = tsdberrors.MultiError{}
-		filesMap = map[storepb.PartialResponseStrategy][]string{}
+		filesMap = map[storepb.PartialResponseStrategy][]string{
```

Comment:
Hm I think that will work, but to make it verbose, clearer and bit more efficient I would suggest add one more loop at the end:

```go
	for s, updater := range *m {
		if _, ok := filesMap[s]; ok {
			// We already updated those above.
			continue
		}
		// Rules removed update updater if needed.
		if len(updater.RuleGroups()) == 0 {
			continue
		}

		if err := updater.Update(evalInterval, []string{}, nil); err != nil {
			errs = append(errs, err)
		}
	}
```
What do you think? (:

Reply:
> Hm I think that will work, but to make it verbose, clearer and bit more efficient I would suggest add one more loop at the end:
> 
> ```go
> 	for s, updater := range *m {
> 		if _, ok := filesMap[s]; ok {
> 			// We already updated those above.
> 			continue
> 		}
> 		// Rules removed update updater if needed.
> 		if len(updater.RuleGroups()) == 0 {
> 			continue
> 		}
> 
> 		if err := updater.Update(evalInterval, []string{}, nil); err != nil {
> 			errs = append(errs, err)
> 		}
> 	}
> ```
> 
> What do you think? (:

it 's more clear. how about merge the last loop with your loop?